docs/integrations/app-integrations/microsoft-teams.mdx

```diff
@@ -7,53 +7,162 @@ In this section, we present how to connect Microsoft Teams to MindsDB.
 
 <Note>
-Read about creating an Incoming Webhook in Microsoft Teams [here](https://learn.microsoft.com/en-us/microsoftteams/platform/webhooks-and-connectors/how-to/add-incoming-webhook?tabs=dotnet).
+Microsoft Entra ID was previously known as Azure Active Directory (Azure AD).
 </Note>
 
+The parameters given above can be obtained by registering an application in Entra ID by following these steps,
+1. Go to the [Azure Portal](https://portal.azure.com/#home) and sign in with your Microsoft account.
+2. Locate the **Microsoft Entra ID** service and click on it.
+3. Click on **App registrations** and then click on **New registration**.
+4. Enter a name for your application and select the **Accounts in this organizational directory only** option for the **Supported account types** field.
+5. Keep the **Redirect URI** field empty and click on **Register**.
+6. Copy the **Application (client) ID** and paste it as the `client_id` parameter, and copy the **Directory (tenant) ID** and paste it as the `tenant_id` parameter.
+7. Click on **Certificates & secrets** and then click on **New client secret**.
+8. Enter a description for your client secret and select an expiration period.
+9. Click on **Add** and copy the generated client secret and paste it as the `client_secret` parameter.
+10. Click on **Authentication** and then click on **Add a platform**.
+11. Select **Web** and enter the following URLs in the **Redirect URIs** field:
+    - `https://cloud.mindsdb.com/verify-auth`
+    - `http://localhost:47334/verify-auth` (for local development)
+
+<Tip>
+You can find more information about creating app registrations [here](https://docs.microsoft.com/en-us/graph/auth-register-app-v2).
+</Tip>
+
+When the above is statement is executed with the given parameters, the handler will open a browser window and prompt you to sign in with your Microsoft account. 
+
+The handler will then act (via the app registration) as the signed in user and will submit requests to the Microsoft Graph API. This is done using the concept of [delegated permissions](https://docs.microsoft.com/en-us/graph/auth/auth-concepts#delegated-permissions).
+
 To connect to Microsoft Teams using MindsDB, the following CREATE DATABASE statement can be used:
 
 ```sql
 CREATE DATABASE teams_datasource
 WITH ENGINE = 'teams',
 PARAMETERS = {
-  "webhook_url": "https://..."
+  "client_id": "your-client-id",
+  "client_secret": "your-client-secret",
+  "tenant_id": "your-tenant-id"
 };
 ```
 
 <Tip>
-If you installed MindsDB locally via pip, you need to install all handler dependencies manually. To do so, go to the handler's folder (mindsdb/integrations/handlers/ms_teams_handler) and run this command: `pip install -r requirements.txt`.
+If you installed MindsDB locally, make sure to install all Microsoft Teams dependencies by running `pip install .[teams]` or from the [requirements.txt](indsdb/integrations/handlers/ms_teams_handler/requirements.txt) file.
```

Comment:
pip install .[teams] is for the zsh, I guess it is better to document the pip install mindsdb[teams]

Reply:
@ZoranPandovski Done.